### DISCUSSION TITLE
(Text) input params for styling?

### DISCUSSION THREAD

**Discussion by vymao** [mantinedev/mantine#3241] [Q&A]
I am trying to create default styling for components (ie. buttons, inputs, boxes). In the [docs](https://mantine.dev/styles/styles-api/#styles-api-with-mantineprovider), I am able to pass params to the button styling via `ButtonStyleParams`: 
```
Button: {
            // Subscribe to theme and component params
            styles: (theme, params: ButtonStylesParams) => ({
              root: {
                height: 42,
                padding: '0 30px',
                backgroundColor:
                  params.variant === 'filled'
                    ? theme.colors[params.color || theme.primaryColor][9]
                    : undefined,
              },
            }),
```
But I don't see this for text inputs. Is this possible?

**Comment by rtivital** [comment]
If type is not exported from the package, you should not use params in MantineProvider styles – there will be inconsistencies.
